### PR TITLE
Test for issue #1977

### PR DESCRIPTION
**Pull request by LefterisJP**
Test that reproduces issue #1977 and will fail on ArchLinux with
Boost version: 1.58.0-1
gcc version: 4.9.2-4

I am opening a PR with it just to enable other people who want to test if they can reproduce the issue on their system.

### CODE REVIEW

**Comment by chfast** [comment]
It is not a VM test. There is a "BasicTests" suite in test/libdevcrypto.

**Comment by chriseth** [comment]
If I see that correctly, libdevcrypto does not even depend on boost's bigints.
I would say either leave it where it is or create a new test suite...

**Comment by chfast** [comment]
Yeah. There is mess there. We don't have basic unit tests. Let just keep
the mess out of the VM part of the testeth.

**Comment by chriseth** [comment]
Why not put it in the libdevcore folder? libdevcore/Common.h defines u256, so I think a good fit would be a test suite called something like "externalDependencies" inside the libdevcore test folder.

**Comment by chfast** [comment]
I agree with "externalDependencies". If you do not have time for that I can do it later.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,7 @@ endmacro()
 
 add_subdirectory(fuzzTesting)
 add_subdirectory(libdevcore)
+add_subdirectory(libdevcore/external_dependencies)
```
Comment (CJentzsch) [inline test/CMakeLists.txt]: rather do a `add_subdirectory(external_dependencies)` in `libdevcore/CMakeLists.txt`

Comment (chfast) [inline test/CMakeLists.txt]: I don't think we use unserscore in folder/file names anywhere.

File: test/libdevcore/external_dependencies/boost.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file boost.cpp
+ * @author Lefteris Karapetsas <lefteris@ethdev.com>
+ * @date 205
```
Comment (chfast) [inline test/libdevcore/external_dependencies/boost.cpp]: Does not look like a date.

File: test/libdevcore/external_dependencies/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+cmake_policy(SET CMP0015 NEW)
+
+aux_source_directory(. SRCS)
```
Comment (chfast) [inline test/libdevcore/external_dependencies/CMakeLists.txt]: This is not the best way of adding files to CMake. The best is to just list all files.

Comment (LefterisJP) [inline test/libdevcore/external_dependencies/CMakeLists.txt]: This is the way that's used all over the tests

File: test/libdevcore/external_dependencies/boost.cpp (hunk 2 above)
Comment (LefterisJP) [inline test/libdevcore/external_dependencies/boost.cpp]: indeed it does not :)

**Pull request merged**